File: PULSE.Services/Database/Servicing.cs
using System;
using System.Collections.Generic;

namespace PULSE.Services.Database
{
    public partial class Servicing
    {
        public Servicing()
        {
            ServicingParts = new HashSet<ServicingPart>();
        }

        public int ServicingId { get; set; }
        public string? ServicingNumber { get; set; }
        public int? CustomerId { get; set; }
        public int? PaymentId { get; set; }
        public string? CustomerRequest { get; set; }
        public int? Status { get; set; }
        public bool? OnlinePayment { get; set; }
        public decimal? LabourCost { get; set; }
        public string? OfferDetails { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public virtual Customer? Customer { get; set; }
        public virtual Payment? Payment { get; set; }
        public virtual ICollection<ServicingPart> ServicingParts { get; set; }
    }
}

File: pulse_mobile/lib/model/servicing/servicing.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_mobile/model/abstract/base_model.dart';
import 'package:pulse_mobile/model/servicing/servicing_part.dart';

part 'servicing.g.dart';

@JsonSerializable()
class Servicing extends BaseModel {
  String? servicingNumber;
  int? customerId;
  int? paymentId;
  String? customerRequest;
  int? status;
  String? statusName;
  bool? onlinePayment;
  double? labourCost;
  String? offerDetails;

  List<ServicingPart> servicingParts = [];

  Servicing();

  factory Servicing.fromJson(Map<String, dynamic> json) =>
      _$ServicingFromJson(json);

  Map<String, dynamic> toJson() => _$ServicingToJson(this);
}
